app/models/batch_import.rb

```diff
@@ -81,6 +82,66 @@ def csv_without_successful_imports
     new_csv
   end
 
+  def self.csv_file_to_hierarchical_json_hash(csv_file)
+    JsonCsv.csv_file_to_hierarchical_json_hash(csv_file.path) do |json_hash_for_row, csv_row_number|
+      # Convert csv-formatted, header-derived json to the expected attribute format
+      BatchImport.import_json_to_digital_object_attribute_format!(json_hash_for_row)
+      yield json_hash_for_row, csv_row_number
+    end
+  end
+
+  # Reads the csv data from the given blob and performs "pre-validation" on the CSV data.
+  # "Pre-validation" is a non-comprehensive check that's meant to quickly catch common errors
+  # without doing more computationally expensive operations like creating objects or doing extensive
+  # database queries.  This method is primarily used for validating a CSV before it's split up and
+  # turned into DigitalObjectImports, which have more extensive checks done as they're processed.
+  #
+  # @return Array(Boolean, String[])
+  #   Sample return value for failed validation:
+  #   [false, ['Something went wrong!', 'Something else went wrong!']]
+  #   Sample return value for successful validation:
+  #   [true, []]
+  def self.pre_validate_blob(blob)
+    pre_validation_errors = []
+    blob.open do |file|
+      # TODO: Make sure there aren't any duplicate headers
+      csv_file_to_hierarchical_json_hash(file) do |json_hash_for_row, _csv_row_number|
+        # TODO: Make sure that only valid fields are present in the dynamic field data properties
```

Comment:
This work should be ticketed, I don't think it's represented anywhere.

Reply:
HYACINTH-602

Reply:
thanks!